File: w1/d1/algo-deaf-grandma/deaf_grandma.js
function deafGrandma() {
  let running = true;
  let goodbye = 0;
  while (running) {
    let x = prompt();
    x = x.toString();

    if (x === 'GOODBYE!') {
      if (goodbye < 1) {
        console.log('LEAVING SO SOON?');
        goodbye++;
      } else {
        console.log('LATER, SKATER!');
        running = false;
      }
    } else if (x == '') {
      console.log('WHAT?!');
    } else if (x.toUpperCase() === x && !Number.isInteger(+x)) {
      console.log('NO, NOT SINCE 1964!');
    } else {
      console.log('SPEAK UP, KID!');
    }
  }
}

deafGrandma();

File: w1/d1/algo-deaf-grandma/deaf_grandma.py
def deaf_grandma():
    print('HEY KID')
    running = True
    goodbye = 0
    while running:
        x = input()
        if x == 'GOODBYE!':
            if goodbye == 1:
                print('LATER, SKATER!')
                running = False
            else:
                print('LEAVING SO SOON?')
                goodbye += 1
        elif x.isupper():
            print('NO, NOT SINCE 1964!')
        elif x == '':
            print('WHAT?!')

        else:
            print('SPEAK UP, KID!')


deaf_grandma()
